### DISCUSSION TITLE
Changing the name of a SharePoint, after PnP Search v4 webparts are configured and working perfectly.

### DISCUSSION THREAD

**Discussion by ccoley0473** [microsoft-search/pnp-modern-search#3193] [Q&A]
October 2022, I was tasked with building a better way to search for documents in a SharePoint document library. I work for a very large outsourced medical billing company and we have various procedures, state regulations, various processes etc our employees must reference daily. Our Compliance and Procedure document library is huge, so the work flow departments just need a better way to sort, filter and search on the documents' Metadata tags and Document titles to find what they need as quickly as possible. Key terms, Key phrases, Managed Columns for the Search Filters, it works perfectly. Instead of hundreds of documents to sort through, now they can narrow the results to less than 5. Upper management is happy and they want to go live with this asap.

I setup the PnP Search web parts just to show the capabilities and proof of concept, I have it working and the user acceptance testing is 100% complete. They have asked if we can change the name of the SharePoint that was used to set this up. They just want to change the Title of the SharePoint from 'Library Hub (Client Service Compliance)' to 'ZOCA'

I feel like changing the name of the SharePoint and / Or the URL will impact the underlying Metadata Term Store, Search Schema, the Query template, Crawled values, RefinableString values, Managed columns and especially the Title column mappings to get it working. But -- I could be wrong.

My question to all -- If I change the name of the SharePoint or URL, do you agree? PnP Search would need to be reconfigured in various places, or would it all just magically adjust?

![Zoca Main page](https://github.com/microsoft-search/pnp-modern-search/assets/141053241/41c2665f-fee3-4e5e-b2a2-a6ea50a2dbc9)

**Comment by wobba** [comment]
Depends on how you have set it up. If you are using the path/URL as part of the query, then you have to change. If not, then you should be able to rename the site title and URL without any issue. Could take some time for re-indexing to catch up, but should work. So I would do this late on a Friday to ensure things are likely to work on Monday. (If content exists on the site you rename the title/URL of).
